File: leetcode/78/78-a.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
       int n = nums.size();
       vector<vector<int>> ans;
       for(int mask = 0; mask < (1 << n); mask++){
           vector<int> cur;
           for(int i = 0; i < nums.size(); i++){
               if((1 << i) & mask){
                   cur.push_back(nums[i]);
               }
           }
           ans.push_back(cur);
       } 
       return ans;
    }
};

File: leetcode/78/78-a.rs
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut res: Vec<Vec<i32>> = Vec::new();
        res.push(vec![]);
        nums.into_iter().for_each(|v|{
            let sz = res.len();
            for i in 0..sz {
                res.push(res[i].clone());
                res.last_mut().expect("should be at least one").push(v);
            }
        });
        res
    }
}